ibis/pyspark/tests/test_timecontext.py

```diff
@@ -13,3 +18,213 @@ def test_table_with_timecontext(client):
     expected = table.compile().toPandas()
     expected = expected[expected.time.between(*context)]
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    ('ibis_window', 'spark_range'),
+    [
+        param(
+            ibis.trailing_window(
+                preceding=ibis.interval(hours=1),
+                order_by='time',
+                group_by='key',
+            ),
+            (-3600, 0),
+        ),  # 1h back looking window
+        param(
+            ibis.trailing_window(
+                preceding=ibis.interval(hours=2),
+                order_by='time',
+                group_by='key',
+            ),
+            (-7200, 0),
+        ),  # 2h back looking window
+        param(
+            ibis.range_window(
+                preceding=0,
+                following=ibis.interval(hours=1),
+                order_by='time',
+                group_by='key',
+            ),
+            (0, 3600),
+        ),  # 1h forward looking window
+    ],
+)
+def test_window_with_timecontext(client, ibis_window, spark_range):
```

Comment:
Can you add a test with has a window op (requires context adjustment) and a non window op (doesn't require context adjustment) inside the same mutate?

Reply:
I think this is a good candidate for test_complex_window, I added it there but found another bug in chain mutate. Create https://github.com/ibis-project/ibis/issues/2453 to track.